src/work/Work.cpp

```diff
@@ -61,7 +61,12 @@ Work::onRun()
     else
     {
         CLOG(DEBUG, "Work") << "Running " << getName();
-        return doWork();
+        auto state = doWork();
+        if (state == State::WORK_SUCCESS)
+        {
+            clearChildren();
```

Comment:
* `clearChildren` was added because after a child goes into WORK_SUCCESS state, it might not be cleaned up by `yieldNextRunningChild` (e.g. if all children succeed on the first iteration, then parent runs and returns success). It shouldn't be dangerous to clear children when parent work returned success (I don't think parent should ever return success if not all children are successful, do you think otherwise?)
* The above describes a successful scenario; However, if a child failed, I think we should shutdown the rest of children like you described (later, abort will help with process handling)